apis/air_quality/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#![no_std]
+
+use core::cell::Cell;
+use libtock_platform::{
+    share::scope, share::Handle, DefaultConfig, ErrorCode, Subscribe, Syscalls,
+};
+use Value::{Tvoc, CO2};
+
+#[cfg(test)]
+mod tests;
+
+enum Value {
+    CO2 = READ_CO2 as isize,
+    Tvoc = READ_TVOC as isize,
+}
+
+pub struct AirQuality<S: Syscalls>(S);
+
+impl<S: Syscalls> AirQuality<S> {
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32,)>>,
```

Comment:
I think we did this wrong here. We need to be able to register a listener that the user can actually use outside this crate, meaning that the listener needs to have a function that is called when the upcall is fired. 

Something similar to the [`ButtonListener`](https://github.com/tock/libtock-rs/blob/cff868b1fb2641f2d1b0f3edaae827cbcfdadef3/apis/buttons/src/lib.rs#L124) should work.

Reply:
I have implemented this listener in my first commit, I will add it again.

Reply:
Yes, that was my mistake :(